proxy/src/main/java/com/velocitypowered/proxy/connection/MinecraftConnection.java

```diff
@@ -85,6 +129,14 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         if (ctx.channel().isActive()) {
+            if (pluginsSessionHandlers != null && !pluginsSessionHandlers.isEmpty()) {
```

Comment:
You need to synchronize this if condition to guarantee memory visibility. I assume methods such as `.write` could get called from different threads.

Reply:
https://github.com/VelocityPowered/Velocity/pull/103/files#diff-98293b855c687bdaddb2c9b78e541088R168
there is a synchronizedList